### ISSUE TITLE
BatList.fold_while

### ISSUE THREAD

**Issue by UnixJunkie** [ocaml-batteries-team/batteries-included#888]
I think we miss this function.
It should do the fold as long as the given predicate holds, and return the unconsumed
part of the list once it does not anymore.

**Comment by UnixJunkie** [comment]
This is not the first time that I need this.
There is one in containers, for example.
I think there is in haskell too IIRC.
If other maintainers are interested, I can send some code.

**Comment by gasche** [comment]
Sure, why not. What's the function signature (can you link to the Contains version)?

**Comment by UnixJunkie** [comment]
The signature in containers (from here: https://c-cube.github.io/ocaml-containers/last/containers/CCList/index.html):
```
val fold_while : ('a -> 'b -> 'a * [ `Stop | `Continue ]) -> 'a -> 'b t -> 'a
```
I just provide it for info.
I don't think it is explicit enough.

**Comment by UnixJunkie** [comment]
My proposal is here:
https://github.com/ocaml-batteries-team/batteries-included/pull/889

**Comment by UnixJunkie** [comment]
PR was merged
